### PR TITLE
Improve Tab system of the lao Activity

### PR DESCRIPTION
**Pull request by GabrielFleischer**
Refactor the tab system in the Lao to be more centralized.

It resembles the change done in #1185

### CODE REVIEW

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java [hunk 1]
```diff
@@ -129,9 +129,25 @@ private void openIdentityMenu() {
...
-        getSupportFragmentManager(), R.id.fragment_lao_detail, LaoDetailFragment::newInstance);
+        getSupportFragmentManager(), R.id.fragment_witnessing, WitnessingFragment::newInstance);
+  }
+
+  private void openDigitalCashMenu() {
+    startActivity(
+        DigitalCashActivity.newIntent(
+            this,
+            mViewModel.getCurrentLaoValue().getId(),
+            mViewModel.getCurrentLaoValue().getName()));
```
Comment (Warhedgehog) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java:142]: We should change it in a future PR if we have time : it's totally useless to provide the name as well

**Review by Warhedgehog — APPROVED** [review]
LGTM. The enum for menu item is a great idea

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java [hunk 2]
```diff
@@ -83,43 +83,43 @@ private void setupBackButton() {
...
+      default:
+        Log.w(TAG, "Unhandled tab type : " + tab);
+    }
   }
 
-  private void openWitnessMenu() {
+  private void openEventsMenu() {
     setCurrentFragment(
-        getSupportFragmentManager(), R.id.fragment_witnessing, WitnessingFragment::newInstance);
+        getSupportFragmentManager(), R.id.fragment_lao_detail, LaoDetailFragment::newInstance);
```
Comment (pierluca) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java:122]: Are you implicitly saying the `LaoDetailFragment` should be renamed LaoEventsFragment ? 😏

**Review by pierluca — APPROVED** [review]

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java (hunk 2 above)
Comment (Warhedgehog) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java:122]: It should

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoTab.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import androidx.annotation.IdRes;
+
+import com.github.dedis.popstellar.R;
+
+import java.util.Arrays;
+import java.util.List;
+
+/** Enum where each element represent a tab in LaoActivity */
+public enum LaoTab {
+  EVENTS(R.id.lao_detail_event_list_menu),
```
Comment (jbsv) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoTab.java:12]: maybe worth having a 'default' tab, that is just an alias of one of these tabs ?

Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoTab.java:12]: I am not certain I understand what you mean by that. Could you be more precise ?

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java (hunk 2 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java:122]: I must have done that subconsciously. But it's a good proposition I made.

**Pull request merged**